propolis/src/common.rs

```diff
@@ -208,6 +204,13 @@ pub struct WriteOp<'a> {
     read_offset: usize,
 }
 impl<'a> WriteOp<'a> {
+    /// Initializes a new write operation from a buffer.
+    ///
+    /// # Arguments
+    ///
+    /// - `op_offset`: An auxiliary offset stored within the operation,
+    /// identifying the region where `buf` should be stored.
```

Comment:
The `op_offset` does not at all relate to an offset within `buf`.  It's the offset of the operation relative to the emulated resource, not the underlying storage (hence why it can be manipulated in `new_child` and such)